### PR TITLE
CLI Tool

### PR DESCRIPTION
**Pull request by zbazztian**
@johnlugton As mentioned, this splits up the GitHub and JIRA functions into their own modules. It also introduces a basic CLI tool. Here are a few examples:
* Synchronize a GitHub repo and a JIRA project:
```
./gh2jira sync  \
  --gh-url https://api.github.com \
  --gh-user zbazztian \
  --gh-token <ghtoken> \
  --gh-org ghas-performance-testing \
  --gh-repo newrepository \
  --jira-url https://jira.demo-stack.com  \
  --jira-user zbazztian \
  --jira-token <jira-password> \
  --jira-project JIR
```
* install the JIRA webhook:
```
./gh2jira hooks install \ 
  --hook-url https://something.com \ 
  --hook-secret <some-secret> \
  --jira-url https://jira.demo-stack.com \ 
  --jira-user zbazztian \
  --jira-token <jira-password>
```
* install the GitHub webhook for a repository (omit the --gh-repo param if webhook should be for entire organization)
```
./gh2jira hooks install \
  --gh-url https://api.github.com \
  --gh-user zbazztian \
  --gh-token <github-token> \
  --gh-org ghas-performance-testing \
  --gh-repo newrepository
```
* Lists existing webhooks:
```
./gh2jira hooks list  \
  --gh-url https://api.github.com \
  --gh-user zbazztian \
  --gh-token <github-token> \ 
  --gh-org ghas-performance-testing \
  --gh-repo newrepository \
  --jira-url https://jira.demo-stack.com \ 
  --jira-user zbazztian \
  --jira-token <jira-password>
```

It roughly works, but needs a lot more polishing. I will amend this PR tomorrow. Of course, if you are working on it today or are testing it, feel free to amend your improvements.

### CODE REVIEW

File: cli.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import json
+import util
+
+
+def fail(msg):
+    print(msg)
+    sys.exit(1)
+
+
+def serve(args):
```
Comment (johnlugton) [inline cli.py:35]: Looks like this is missing an implementation...

**Comment by johnlugton** [comment]
I tested sync and seems to be working well with these changes: https://github.com/johnlugton/lgtm-issue-tracker-example/pull/6

overall LGTM - thanks for all your work on this!

**Comment by johnlugton** [comment]
We should also pick a repo where this is going to live longer term and that we can make available to customers. @zbazztian do you have a preference? Not sure if field architecture have somewhere for this...

cc @cmboling

File: cli.py (hunk 1 above)
Comment (zbazztian) [reply cli.py:35]: Agreed, I hope to find some time today to polish the CLI a bit more.

**Comment by zbazztian** [comment]
@johnlugton Thanks for testing this. I will polish this a bit more during the remainder of this week and also check how it works with the new Code Scanning API. The synchronization code isn't very optimized yet, because it fires quite a few API requests during a full sync, which might be less than optimal for larger setups with many repositories. I have concrete ideas for how to make it faster, but the focus until now was just on making it work.

I don't have a preference for where it is going to be hosted. Do you know if there a preferred location / organization for these types of open source repositories which will be used by customers? Perhaps somewhere under the [github organization](https://github.com/github/)?

**Comment by cmboling** [comment]
@zbazztian I think it would make sense to move this over to the `github` org

**Comment by zbazztian** [comment]
@cmboling Agreed. I'll clean up a few more things before I'll make the move.

**Comment by zbazztian** [comment]
@johnlugton This is becoming a bit of a kitchen sink. Can we merge this, so I can create another PR with the rest of the fixes?

**Pull request merged**